wire-ios-sync-engine/Source/UserSession/ZMUserSession/ZMUserSession.swift

```diff
@@ -424,6 +427,12 @@ public final class ZMUserSession: NSObject {
         applicationStatusDirectory.syncStatus.syncStateDelegate = self
         applicationStatusDirectory.clientRegistrationStatus.registrationStatusDelegate = self
 
+        let observeMLSGroupVerificationStatusUseCase = observeMLSGroupVerificationStatusUseCase ?? ObserveMLSGroupVerificationStatusUseCase(
+            mlsService: mlsService,
+            updateMLSGroupVerificationStatusUseCase: updateMLSGroupVerificationStatus,
+            syncContext: coreDataStack.syncContext
+        )
+
```

Comment:
Is this only injected for tests? The tests also seem to just pass an empty implementation, suggesting that we don't want any observation during tests.

I feel like `ObserveMLSGroupVerificationStatusUseCase` shouldn't be a use case, but rather just an object that observes epoch changes and invokes another use case. I'm wondering if we even need to have this here and instead setup and tear down the observation when needed.

Reply:
1. Yes it's only for tests. My assumption is that the original idea was tests using the fat `ZMUserSession` with all kinds of side effects should not perform them and thus causing suddenly unexpected behaviour.

2. I agree that it's not really a use case. The reason why it lives in `ZMUserSession` is probably the life cycle and that it must be retained as an observer. @David-Henner maybe you can give us some additional insight?

Reply:
@johnxnguyen I worked on this and chose a different solution to the use case. Would be happy to get some feedback :) thanks